### PR TITLE
TE-3820: Changed structure of collector

### PR DESCRIPTION
**Pull request by supproduction**
https://spryker.atlassian.net/browse/TE-3820

### CODE REVIEW

File: src/collector/wrappers/index.ts [hunk 1]
```diff
@@ -4,25 +4,21 @@ import { IParsedViewsResult } from '../views';
...
+        [key: string]: IParsedModulesByName
     }
 }
 
+interface IParsedModulesByName {
+    template?: IParsedTemplates[]
+    molecule?: IParsedComponent[]
+    atom?: IParsedComponent[]
+    organism?: IParsedComponent[]
+    view?: IParsedViewsResult[]
```
Comment (KostKravchenko) [inline src/collector/wrappers/index.ts]: Please rename interface to IParsedViews.

File: src/collector/views/index.ts [hunk 2]
```diff
@@ -42,10 +42,10 @@ const limitedScanForFilesCollection = scanForViews => iif(
 export const getObservable = (): Observable<IParsedViewsResult> => restrictedScanForTwigsCollection.pipe(
     mergeMap(scanForViews =>
         limitedScanForFilesCollection(scanForViews).pipe(
-            map(getFile),
+            map(getView),
             tap(debugViews),
             flatMap(parseViews),
-            tap(printParsedFileLog),
+            tap(printParsedTemplateLog),
```
Comment (KostKravchenko) [inline src/collector/views/index.ts]: Please rename to printParsedViewLog

**Review by KostKravchenko — CHANGES_REQUESTED** [review]

File: src/collector/views/index.ts (hunk 2 above)
Comment (supproduction) [reply src/collector/views/index.ts]: It already renamed

File: rules/correct-twig-component-structure.js [hunk 3]
```diff
@@ -7,13 +7,20 @@ module.exports = class extends Rule {
     }
 
     test(data) {
-        parseOutputFieldHelper(data.components).forEach(component => {
+        parseOutputFieldHelper(data.modules).forEach(component => {
             const { files, type, name, path } = component;
+            const errorMessage = (message) => `${message} ${type} ${bold(name)}:\\n${dim(path)}`;
```
Comment (FRITS-Kh) [inline rules/correct-twig-component-structure.js]: Please remove parentheses for `(message)`.

File: src/collector/wrappers/index.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    view?: IParsedViews[]
+}
+
+type TModulePart = IParsedTemplates[] | IParsedComponent[] | IParsedViews[];
+
+export function getModuleWrapper(components, templates, views): IParsedModules {
+    let modules: IParsedModules = {};
+    const moduleCreation = (modulePart: TModulePart) => {
+        const moduleLevels: string[] = Object.keys(modulePart);
+        const modulesByName = (modulesByLevel, moduleLevel) =>  modulesByLevel.forEach((moduleByLevel) => {
```
Comment (FRITS-Kh) [inline src/collector/wrappers/index.ts]: Please remove parentheses for `(moduleByLevel)`

**Review by FRITS-Kh — CHANGES_REQUESTED** [review]

File: src/collector/components/component.ts [hunk 5]
```diff
@@ -61,27 +63,30 @@ function extractCorrectScriptFileName(defaultFileName: string, directoryFiles: s
     return defaultFileName;
 }
 
-export function getComponent(path: string): IComponent {
-    const name = basename(path);
-    const deprecated = getFile(join(path, 'DEPRECATED.md'));
-    const directoryFiles = readdirSync(path);
+export function getComponent(level: TLevelRestriction) {
+    return function (path: string): IComponent {
```
Comment (FRITS-Kh) [inline src/collector/components/component.ts]: Please avoid closure in functions like this.

**Review by FRITS-Kh — CHANGES_REQUESTED** [review]

**Review by FRITS-Kh — APPROVED** [review]

**Review by KostKravchenko — APPROVED** [review]

**Pull request merged**